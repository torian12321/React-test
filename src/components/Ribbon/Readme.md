#### Basic
```
<Panel>
  <Ribbon>Ribbon</Ribbon>
</Panel>
```

#### Multiple Ribbons
```
<Panel>
  <Ribbon>Ribbon</Ribbon>
  <Ribbon>second Ribbon</Ribbon>
</Panel>
