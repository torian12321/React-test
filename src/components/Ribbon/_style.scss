@import '../../styles/conf/index.scss';

// Vars
$ribbon-offset        : 1em !default;
$ribbon-font-size     : $font-size !default;
$ribbon-color-main    : $themeColor_1 !default;
$ribbon-border-radious: $border-radious !default;

// Mixins
@mixin ribbon-palette($color: $ribbon-color-main) {
  background: $color;
  @include smart-text-color($color);

  &:after {
    border-top-color: darken($color, 10%);
  }
}

// Block
.ribbon {
  position   : absolute;
  font-size  : $ribbon-font-size;
  right      : -$ribbon-offset;
  min-width  : 48px;
  padding    : .4em 1em;
  line-height: 1em;

  @include rounded($ribbon-border-radious 0 0 $ribbon-border-radious);

  // Create distance between different ribbons
  @for $index from 1 through 10 {
    &:nth-child(#{$index}) {
      top: calc(-1em + (#{$index} * 2.4em));
    }
  }

  // arrow
  &:after{
    $size: $ribbon-offset;

    content     : '';
    position    : absolute;
    top         : 100%;
    right       : 0;
    width       : 0;
    height      : 0;
    background  : transparent;
    border      : solid transparent;
    border-width: $size $size 0 0;
  }

  // Colors
  @include ribbon-palette($ribbon-color-main);
}