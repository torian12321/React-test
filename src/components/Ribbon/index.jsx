import React      from 'react';
import PropTypes  from 'prop-types';
import classNames from 'classnames';
import './_style.scss';

class Ribbon extends React.Component {
  render() {
    return(
      <div className={classNames(
        'ribbon',
        this.props.className
      )}>
        {this.props.children}
      </div>
    )
  }
}

Ribbon.propTypes = {
  children : PropTypes.node.isRequired,
  className: PropTypes.string
};

module.exports = Ribbon;
