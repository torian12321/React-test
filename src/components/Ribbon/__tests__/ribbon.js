import React    from 'react';
import renderer from 'react-test-renderer';
import Ribbon   from '../index';

describe('component', () => {
  describe('<Ribbon />', () => {
    
    it('should render correctly', () => {
      expect(renderer.create(
        <Ribbon>Lorem ipsum</Ribbon>
      )).toMatchSnapshot();
    });

  });
});
