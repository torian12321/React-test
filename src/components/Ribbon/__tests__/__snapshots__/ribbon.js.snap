// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <Ribbon /> should render correctly 1`] = `
<div
  className="ribbon"
>
  Lorem ipsum
</div>
`;
